### ISSUE TITLE
New/Updated Icon Artwork

### ISSUE THREAD

**Issue by zorgiepoo** [zorgiepoo/Bit-Slicer#18]
Bit Slicer's current application and user document icons don't scale very well with small sizes, and isn't up to date with the "post-Yosemite appearance."

The current application icon conveys slicing bits, which is analogous to messing around with a game or application, I guess. The idea could stay the same, but it doesn't have to.

Relevant must-reads: 
["OS X Human Interface Guidelines - Icon and Image Design"](https://developer.apple.com/library/mac/documentation/UserExperience/Conceptual/OSXHIGuidelines/Gallery.html#//apple_ref/doc/uid/20000957-CH88-SW1)

**Comment by etybyte** [comment]
Just threw together a quick 5 minute thing based on the current icon. It's not great. ![bitslicericon](https://cloud.githubusercontent.com/assets/9469380/8692437/27963cc6-2a83-11e5-9730-46e8503231f0.png)

**Comment by etybyte** [comment]
or better yet perhaps something more like this... maybe with a slightly smaller split and a couple more numbers.
![bitslicericon2](https://cloud.githubusercontent.com/assets/9469380/8692969/3d63841e-2a88-11e5-84a7-3fbff75d6b5b.png)

**Comment by zorgiepoo** [comment]
The idea is similar to the current icon, but I'd prefer to have some perspective and real user identifiable objects. This mockup is lacking in detail/interest. A circular icon like iTunes and App Store is probably not very desirable for this kind of utility. Additionally, not too much emphasis should be put on text. I appreciate the effort.

**Comment by zigmasdirigeant** [comment]
**Hi Zorgie!**
Today I had a while to play around with the App icon, I started with some basic concepts already know from apps inside the OS. After that I evolved the idea until I had some memorable and likeable. I already read on your last comment that you want some _User identificable objects_ and that _”A circular icon like iTunes and App Store is probably not very desirable for this kind of utility”_ I suppose because you are trying to unmark from the more standard icons, but I think that we can find something nice even with or without identificable objects or even using the round mac icon.

I’ll leave you here 5 different icons that I’m working on and I’ll love to hear your opinion.

**Option 1:** Based on the regular circle app icon but with a twist. Trying to find a graphical way to slice a byte, while maintaining a minimal look. It almost looks like a shield being sliced, a good way to represent what the app does to the programs.

![bit slicer - option 1 title-80](https://user-images.githubusercontent.com/2717486/42403026-c42bd168-817e-11e8-941e-208562eb0822.jpg)
![bit slicer - option 1 sizes-80](https://user-images.githubusercontent.com/2717486/42403040-d3670d32-817e-11e8-862d-bd3acb3f1641.jpg)
![bit slicer - option 1 bar-80](https://user-images.githubusercontent.com/2717486/42403039-d0ac877a-817e-11e8-86e7-9078dc41d9ed.jpg)

<br><br>
**Option 2:** Same as the last one, but lighter. This icon is the nearest of them all to the standard OS icon, taking the same twist and adding elegancy to the style.

![bit slicer - option 2 tittle-80](https://user-images.githubusercontent.com/2717486/42403153-8c105956-817f-11e8-94f5-ccb5dbe00aec.jpg)
![bit slicer - option 2 sizes-80](https://user-images.githubusercontent.com/2717486/42403164-974e8220-817f-11e8-8d45-31b7ed870759.jpg)
![bit slicer - option 2 bar-80](https://user-images.githubusercontent.com/2717486/42403157-8fa7a70e-817f-11e8-89ea-58392374284b.jpg)

<br><br>
**Option 3:** One of the basic concepts you have to grasp to modify any memory is *HEX*. That’s why i searched a way to combine hex+bitslicer name+ action. Still in the zone of the mac os icons, but in a different world.
![bit slicer - option 3 title-80](https://user-images.githubusercontent.com/2717486/42403341-c0ccbfc6-8180-11e8-8fdf-0f9a64b2f946.jpg)

<br><br>
**Option 4:** ON the range of the hexagon medals, but with the concept of “breaking” a game, we have a sliced steel with the B icon.
![bit slicer - option 4 title-80](https://user-images.githubusercontent.com/2717486/42404503-e0d7b0ba-8189-11e8-8d2d-586b7d6ceb12.jpg)
![bit slicer - option 4 sizes-80](https://user-images.githubusercontent.com/2717486/42404506-e6b5f1d6-8189-11e8-8956-bc0f0e882a0d.jpg)
![bit slicer - option 4 bar-80](https://user-images.githubusercontent.com/2717486/42404507-e98a4dd0-8189-11e8-80bd-c280e6dc04bd.jpg)

<br><br>
**Option 5: The last is maybe a long shot... The action of slicing taken to the edge.

![bit slicer - option 5 title-80](https://user-images.githubusercontent.com/2717486/42404520-13926b12-818a-11e8-8964-d05683c7b287.jpg)
![bit slicer - option 5 sizes-80](https://user-images.githubusercontent.com/2717486/42404523-17cffffa-818a-11e8-855f-61ab8de5fd9e.jpg)
![bit slicer - option 5 bar-80](https://user-images.githubusercontent.com/2717486/42404525-1a9a5190-818a-11e8-9a62-729d7b212471.jpg)

<br><br>
<br><br>
<br><br>
![bitslicer-5-logos](https://user-images.githubusercontent.com/2717486/42404566-7b6b1af4-818a-11e8-9c9c-8a0e69603eee.png)
<br><br>
<br><br>
<br><br>

So!? I hear yah! :)
Cheers!

**Comment by etybyte** [comment]
Well then, now I'm embarrassed by my attempted contributions, haha. Someone with a bit more graphical design talent showed up.

Hm, I personally like the hexagon idea although I feel like Option 4, the "sliced steel" variant, doesn't hold up well enough at small sizes and while it's not all that similar I'm wondering if the solitary "B" is more likely to evoke thoughts of bitcoin and the bitcoin logo, in particular on the variants with a line through it. The bitcoin thing both may not be true and may not be worth worrying about even if it _is_ true.

If I were picking the logo and picking from these five I'd probably go with Option 3, but I'm not sure it conveys the "slicing of bits" concept clearly enough.

The reality regarding perspective is that it's being phased out of icons and if one of the hopes for a new Bit Slicer icon is fitting in and feeling on par with the other icons and the "post-Yosemite appearance" moving away from perspective is a good idea. It seems to me the best fit of the current three main macOS icon styles for Bit Slicer might be the tilted rectangle used as the basis for Script Editor, Calculator,  Font Book, Text Edit, Pages, etc.

Real user identifiable objects is a tough one. The purpose, function, and use of Bit Slicer is very much fundamentally tied to _digital_ constructs and ones that don't really have direct physical parallels in many ways.

**Comment by zigmasdirigeant** [comment]
**Hey Etybyte!**

Not even! You contributions where nice :), thanks!

The thing with the smaller variants is that they can be modified to look well on small sizes. The Mac Os .Icns format let you chose a different source for each size, so in the final build, I already thought of making the ridges on the cracks stronger, so you don’t lose the effect on small resolutions.

YES I encounter that problem when designing the icon, every other font that I liked ended looking too _”Bitcoiny”_ that’s why I decided for this type. More modern and bold, so there was no confusion or relation with the Bitcoin logo, that uses a more clasic serif font... The other solution was using the other initial: **”BS”**... But... Well. I read it like bullshit 👎 XD .

![unknown](https://user-images.githubusercontent.com/2717486/42408899-54b3ee52-81d2-11e8-86a2-a230a4188530.jpeg)

Number 3 is definitely my favourite too, from a graphical point of view. I find it clean and cool.

So, you think I should try to figure out an icon using that style? The tilted rectangle I mean? It weird, but I have this feeling that the tilted rectangle feels old. But It’s just me, if you actually see the new Mohave icons, they maintain a lot the tilted rectangle... So yeah. But also, why don’t we differentiate Bit slicer from the rest? In the end, this styles are only guide lines. 🤔 Hmm... I’ll give it a thought and a shot about this. See what I can get out of it.

Thanks for your comments man! 
Appreciate them.

Cheers!

**Comment by etybyte** [comment]
Yeah, I'm aware of the structure of macOS icons and that's definitely a choice for resolving some of the smaller variants issues, 3 in particular.

Haha yeah both logos and icons, in particular, are really difficult in that regard.

Yeah, BS is a definite no-go, haha.

I was pointing out the tilted rectangle as a potential modern appearance equivalent to the "perspective" that Zorgie was looking for.

**Comment by zorgiepoo** [comment]
Yeah, Bit Slicer's current icon is a tilted rectangle style like Xcode, Notes, Text Edit, Pages, Hopper Disassembler, Hex Fiend. I like that more than a circular icon for this app.

"Post-yosemite" appearance may have meant more a flattened look (it's been a while since I've looked at icons pre Yosemite for the above mentioned apps though). For example, you can either take the current app icon and remove the sheet of paper that's behind, or make the sheet of paper that's behind more aligned rather than being on a different tilt, maybe. You'd still want a little depth/shadow to make the icon not look too awkward.

I don't think rendering a "B" letter is the right way to go. It's English specific and doesn't convey much about the app to me. You'd also have to be careful about not getting distracted by little visual effects like fading/fuzzing and scratch marks.

I actually kind of like something like the current app icon, just not the current quality of it. Anyway, icons are hard :|..

**Comment by etybyte** [comment]
This link has a bunch of the icons pre- and post- Yosemite side-by-side:
[The New Icons of OS X Yosemite – A Side-by-Side Comparison](https://www.tekrevue.com/os-x-yosemite-vs-mavericks-icons/)

A lot of the pre- icons had actual perspective to them, for example, the edges of the cover for Dictionary are not parallel lines in the pre- icon, but the majority of those have been flattened and had the perspective removed. Maybe just try a simpler version of the current icon that's more inline with the other tilted rectangle icons?

![bit_slicer_old_sloppy_modernization_01_128](https://user-images.githubusercontent.com/9469380/42418762-dd71125e-825b-11e8-9e6c-27c3e5a72eae.png)

That's a good point about using a letter in the icon in terms of making it English specific.

Icons are definitely hard, haha.

**Comment by maxtron95** [comment]
Updated the current icon design for modern versions of macOS.
![bit-slicer-update](https://github.com/user-attachments/assets/c7259ae4-9749-48c6-ba74-d4a88355ab0f)
[bit-slicer-updated-icon.zip](https://github.com/user-attachments/files/18016239/bit-slicer-updated-icon.zip)

**Comment by maxtron95** [comment]
Updated the document icon to match modern macOS

![Bit-slicer-file](https://github.com/user-attachments/assets/01242111-d6e6-4ce2-9266-60cfebcceb33)
[Bit-slicer-file.zip](https://github.com/user-attachments/files/18241517/Bit-slicer-file.zip)

**Comment by maxtron95** [comment]
@zorgiepoo I'm fine with just maxtron95. About copyright, I don't mind where it's used really as long as I'm credited.
